src/utils.ts

```diff
@@ -714,10 +715,10 @@ export const getResponseSchemaValidator = (
 	const compile = (schema: TSchema, references?: TSchema[]) => {
 		const cleaner = (value: unknown) => {
 			if (!value || typeof value !== 'object')
-				return Value.Clean(schema, value)
+        return Value.Clean(schema, references || [], value)
 
-			if (Array.isArray(value)) value = Value.Clean(schema, value)
-			else value = Value.Clean(schema, value)
+      if (Array.isArray(value)) value = Value.Clean(schema, references || [], value)
+      else value = Value.Clean(schema, references || [], value)
```

Comment:
Would suggest adhering to @SaltyAom's convention of using tabs for indentation 🙂